spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/intermediate/IRGraph.scala

```diff
@@ -141,7 +141,10 @@ private[bigdl] class IRGraph[T: ClassTag](
       if (input.isInstanceOf[Tensor[T]]) {
         // todo: handle for 3 dimensions, expand 3 dims to 4 dims
         val size = input.toTensor[T].size()
-        val sizeNew = if (size.length == 3)  Array(size(0), 1, size(1), size(2)) else size
+        val sizeNew = if (size.length == 3 && inputFormats(0) != Memory.Format.ntc
+          && inputFormats(0) != Memory.Format.tnc) {
+          Array(size(0), 1, size(1), size(2))
+        } else size
```

Comment:
Why we should create `sizeNew` ? Could it be handled in `Input` layer/op ?

Reply:
input layer needs input shape and format. Dim 3 has no corresponding format, and its actual channel is 1, so here we add one dim and create sizeNew

Reply:
OK. Got it.